common/safety-push-metrics/src/counters.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::atomic::{AtomicBool, AtomicI64, Ordering};
+
+pub struct BaseCounter {
+    counter_name: String,
+    counter_help: String,
+    counter: AtomicI64,
```

Comment:
So this is a "base class" for Gauge and Counter structs. Gauge requires a signed integer.

I'll probably rename this to `BaseMetric` to avoid confusion.